File: unitconverter.java
import java.util.Scanner;
import java.text.DecimalFormat;

public class UnitConverter {

    public static void unitConverter() {
        Scanner scanner = new Scanner(System.in);
        DecimalFormat df = new DecimalFormat("0.00");

        int choice;
        double value;
        double INRtoUSDfctr = 0.011;
        double INRtoJPYfctr = 1.73;
        double INRtoEURfctr = 0.011;
        double KGtoLBfctr = 2.204;
        double LBtoKGfctr = 0.453;

        System.out.println("Unit Converter");
        System.out.println("1. Currency");
        System.out.println("2. Temperature");
        System.out.println("3. Mass");
        System.out.print("Enter your choice: ");
        choice = scanner.nextInt();

        switch (choice) {
            case 1:
                System.out.println("Currency Converter");
                System.out.println("1. INR to USD");
                System.out.println("2. INR to JPY");
                System.out.println("3. INR to EUR");
                System.out.print("Enter your choice: ");
                choice = scanner.nextInt();

                switch (choice) {
                    case 1:
                        System.out.print("Enter INR value: ");
                        value = scanner.nextDouble();
                        System.out.println(df.format(value) + " INR = " + df.format(value * INRtoUSDfctr) + " USD");
                        break;
                    case 2:
                        System.out.print("Enter INR value: ");
                        value = scanner.nextDouble();
                        System.out.println(df.format(value) + " INR = " + df.format(value * INRtoJPYfctr) + " JPY");
                        break;
                    case 3:
                        System.out.print("Enter INR value: ");
                        value = scanner.nextDouble();
                        System.out.println(df.format(value) + " INR = " + df.format(value * INRtoEURfctr) + " EUR");
                        break;
                    default:
                        System.out.println("Invalid choice!");
                }
                break;
            case 2:
                System.out.println("Temperature Converter");
                System.out.println("1. Celsius to Fahrenheit");
                System.out.println("2. Fahrenheit to Celsius");
                System.out.print("Enter your choice: ");
                choice = scanner.nextInt();

                switch (choice) {
                    case 1:
                        System.out.print("Enter Celsius value: ");
                        value = scanner.nextDouble();
                        System.out.println(df.format(value) + " Celsius = " + df.format((value * 9 / 5) + 32) + " Fahrenheit");
                        break;
                    case 2:
                        System.out.print("Enter Fahrenheit value: ");
                        value = scanner.nextDouble();
                        System.out.println(df.format(value) + " Fahrenheit = " + df.format((value - 32) * 5 / 9) + " Celsius");
                        break;
                    default:
                        System.out.println("Invalid choice!");
                }
                break;
            case 3:
                System.out.println("Mass Converter");
                System.out.println("1. Kilograms to Pounds");
                System.out.println("2. Pounds to Kilograms");
                System.out.print("Enter your choice: ");
                choice = scanner.nextInt();

                switch (choice) {
                    case 1:
                        System.out.print("Enter kilograms value: ");
                        value = scanner.nextDouble();
                        System.out.println(df.format(value) + " kilograms = " + df.format(value * KGtoLBfctr) + " pounds");
                        break;
                    case 2:
                        System.out.print("Enter pounds value: ");
                        value = scanner.nextDouble();
                        System.out.println(df.format(value) + " pounds = " + df.format(value * LBtoKGfctr) + " kilograms");
                        break;
                    default:
                        System.out.println("Invalid choice!");
                }
                break;
            default:
                System.out.println("Invalid choice!");
        }

        unitConverter(); // Recursive call
    }

    public static void main(String[] args) {
        unitConverter();
    }
}

File: unitconverter.c
#include <iostream>
#include <iomanip>

void unitConverter() {
    int choice;
    float value;
    float INRtoUSDfctr = 0.011f;
    float INRtoJPYfctr = 1.73f;
    float INRtoEURfctr = 0.011f;
    float KGtoLBfctr = 2.204f;
    float LBtoKGfctr = 0.453f;

    std::cout << "Unit Converter\n";
    std::cout << "1. Currency\n";
    std::cout << "2. Temperature\n";
    std::cout << "3. Mass\n";
    std::cout << "Enter your choice: ";
    std::cin >> choice;

    switch (choice) {
        case 1:
            std::cout << "Currency Converter\n";
            std::cout << "1. INR to USD\n";
            std::cout << "2. INR to JPY\n";
            std::cout << "3. INR to EUR\n";
            std::cout << "Enter your choice: ";
            std::cin >> choice;

            switch (choice) {
                case 1:
                    std::cout << "Enter INR value: ";
                    std::cin >> value;
                    std::cout << std::fixed << std::setprecision(2) << value << " INR = " << value * INRtoUSDfctr << " USD\n";
                    break;
                case 2:
                    std::cout << "Enter INR value: ";
                    std::cin >> value;
                    std::cout << std::fixed << std::setprecision(2) << value << " INR = " << value * INRtoJPYfctr << " JPY\n";
                    break;
                case 3:
                    std::cout << "Enter INR value: ";
                    std::cin >> value;
                    std::cout << std::fixed << std::setprecision(2) << value << " INR = " << value * INRtoEURfctr << " EUR\n";
                    break;
                default:
                    std::cout << "Invalid choice!\n";
            }
            break;
        case 2:
            std::cout << "Temperature Converter\n";
            std::cout << "1. Celsius to Fahrenheit\n";
            std::cout << "2. Fahrenheit to Celsius\n";
            std::cout << "Enter your choice: ";
            std::cin >> choice;

            switch (choice) {
                case 1:
                    std::cout << "Enter Celsius value: ";
                    std::cin >> value;
                    std::cout << std::fixed << std::setprecision(2) << value << " Celsius = " << (value * 9 / 5) + 32 << " Fahrenheit\n";
                    break;
                case 2:
                    std::cout << "Enter Fahrenheit value: ";
                    std::cin >> value;
                    std::cout << std::fixed << std::setprecision(2) << value << " Fahrenheit = " << (value - 32) * 5 / 9 << " Celsius\n";
                    break;
                default:
                    std::cout << "Invalid choice!\n";
            }
            break;
        case 3:
            std::cout << "Mass Converter\n";
            std::cout << "1. Kilograms to Pounds\n";
            std::cout << "2. Pounds to Kilograms\n";
            std::cout << "Enter your choice: ";
            std::cin >> choice;

            switch (choice) {
                case 1:
                    std::cout << "Enter kilograms value: ";
                    std::cin >> value;
                    std::cout << std::fixed << std::setprecision(2) << value << " kilograms = " << value * KGtoLBfctr << " pounds\n";
                    break;
                case 2:
                    std::cout << "Enter pounds value: ";
                    std::cin >> value;
                    std::cout << std::fixed << std::setprecision(2) << value << " pounds = " << value * LBtoKGfctr << " kilograms\n";
                    break;
                default:
                    std::cout << "Invalid choice!\n";
            }
            break;
        default:
            std::cout << "Invalid choice!\n";
    }

    unitConverter(); // Recursive call to repeat the process
}

int main() {
    unitConverter(); // Start the recursive function
    return 0;
}
